internal/cmd/cmd.go

```diff
@@ -294,3 +306,50 @@ Additional help topics:{{range .Commands}}{{if .IsAdditionalHelpTopicCommand}}
 Use "{{.CommandPath}} [command] --help" for more information about a command.{{end}}
 `
 }
+
+// warnVersion checks if the client version is ahead of the server and logs a warning if so
+func warnVersion(ctx context.Context, api *api.Client) {
```

Comment:
We'll want to fail outright to avoid unintended consequences that could happen from only warning (e.g. ugly errors)